PsfShimMonitor/KernelTrace.cs

```diff
@@ -0,0 +1,1634 @@
+                                                 lock (_TKernelEventListsLock)
+                                                 {
+                                                     _TKernelEventListItems.Add(ei);
+                                                 }
+                                                 worker.ReportProgress((int)data.EventIndex);
+
+
+                                             }
+                                             catch
+                                             {
+                                                 ;
+                                             }
+                                         }
+                                         else if (data.EventName.StartsWith("Image/Unload"))
+                                         {
+                                             ; // ignore
+                                         }
+                                         else if (data.EventName.StartsWith("FileIOInit"))
+                                         {
+                                             ;  // ignore for now
+                                         }
+                                         else if (data.EventName.StartsWith("FileIO"))
+                                         {
+                                             if (data.EventName.StartsWith("FileIO/Query")) // also catch QueryInfo
+                                             {
+                                                 // FileIO/Query     (ulong)IrpPtr, (ulong)FileObject, (ulong)FileKey, (ulong)ExtraInfo, InfoClass, FileName
+                                                 // FileIO/QueryInfo (ulong)IrpPtr, (ulong)FileObject, (ulong)FileKey, (ulong)ExtraInfo, InfoClass, FileName
+#if DEBUG
+                                                 if (_ModelEventItems.Count < 10000)
+#else
+                                                 if (FilterOnProcessId == pid)
+#endif
+                                                 {
+                                                     try
+                                                     {
+                                                         string inputs = "FileName=\t" + data.PayloadStringByName("FileName");
+                                                         inputs += "\nFileKey=    \t0x" + ((ulong)data.PayloadByName("FileKey")).ToString("x");
+                                                         inputs += "\nExtraInfo=  \t0x" + ((ulong)data.PayloadByName("ExtraInfo")).ToString("x");
+                                                         string outputs = "FileObject=\t0x" + ((ulong)data.PayloadByName("FileObject")).ToString("x");
+                                                         outputs += "\nIrpPtr=    \t0x" + ((ulong)data.PayloadByName("IrpPtr")).ToString("x");
+
+                                                         EventItem ei = new EventItem((int)data.EventIndex,
```

Comment:
lots of repetition around EventItem construction - maybe factor out another ctor that takes a data object directly, or a helper?

Reply:
Addressed in commit 727a9e9856d361f686523f2ea2907876c3ef9b23 on 11/28